crates/sui-core/src/transaction_manager.rs

```diff
@@ -197,6 +197,11 @@ impl AvailableObjectsCache {
         }
     }
 
+    #[cfg(test)]
+    fn new_with_size_for_unittest(metrics: Arc<AuthorityMetrics>, size: usize) -> Self {
```

Comment:
Why not keep the test calling new_with_size() directly?

Reply:
fixed - an earlier version of the code did the randomization in new_with_size so i needed an escape hatch for the tests